src/core/exec-invoke.c

```diff
@@ -3408,6 +3418,7 @@ static int apply_mount_namespace(
                 .incoming_dir = incoming_dir,
                 .private_namespace_dir = private_namespace_dir,
                 .notify_socket = root_dir || root_image ? params->notify_socket : NULL,
+                .notify_socket_path = exec_get_effective_notify_socket_path(context, params, needs_sandboxing),
```

Comment:
And I'd always pass host_notify_socket over. IOW, conditionalize the check of bind mounting notify socket on notify_socket_path instead. As a side effect it would start to properly honor MountAPIVFS= too